lib/freyja/query_service.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module Freyja
+  class QueryService
```

Comment:
I'd :+1: adding a composite query service like this in Valkyrie if you wanted.

Reply:
seems like a good plan to get through it in Hyrax land and then port it. Since I'm modifying a valkyrie branch to make the shared examples flexible enough for this use case, it should slot over super easy if we get to that point.